### PR TITLE
[CMake] Only build SwiftSyntax if building SDK overlays

### PR DESCRIPTION
**Pull request by harlanhaskins**
SwiftSyntax depends on Foundation, which depends on the SDK overlays
being built. However, the existing build configuration tried to build
SwiftSyntax even if the SDK overlays were not built. Ensure we're
building overlays before building SwiftSyntax, and guard tests with an
`sdk_overlay` test.

Resolves [SR-6008](https://bugs.swift.org/browse/SR-6008).

### CODE REVIEW

File: test/lit.cfg [hunk 1]
```diff
@@ -295,6 +295,10 @@ sdk_overlay_linker_opt = ""
 sdk_overlay_dir_opt = ""
 test_sdk_overlay_dir = lit_config.params.get('test_sdk_overlay_dir', None)
 if test_sdk_overlay_dir is not None:
+
+    # TODO: Is this the right place to put this?
+    config.available_features.add('sdk_overlay')
```
Comment (harlanhaskins) [inline test/lit.cfg]: @gottesmm Is this the right place to add the `sdk_overlay` feature?

**Comment by harlanhaskins** [comment]
@davezarzycki This now builds with the "quick-debug" script:

```
time nice ./utils/build-script \                                                 
  --build-subdir="$(dirname $PWD)/d" \                                           
  --llvm-targets-to-build X86 \                                                  
  --skip-ios --skip-tvos --skip-watchos \                                        
  --skip-build-benchmarks \                                                      
  --build-swift-static-stdlib false \                                            
  --build-swift-static-sdk-overlay false \                                       
  --build-swift-dynamic-sdk-overlay false \                                      
  --build-swift-stdlib-unittest-extra false \                                    
  -r \                                                                           
  --debug-swift \                                                                
  --debug-swift-stdlib
```

```
13809.10s user 626.79s system 641% cpu 37:31.85 total
```

**Comment by harlanhaskins** [comment]
@swift-ci please clean test

File: test/lit.cfg (hunk 1 above)
Comment (davezarzycki) [reply test/lit.cfg]: What's blocking this PR from being merged?

Comment (harlanhaskins) [reply test/lit.cfg]: I suppose that's the right place for this. I'll remove the comment and merge.
Ideally we can pair this with a build system unit test. @Rostepher is that still a possibility?

Comment (Rostepher) [reply test/lit.cfg]: I'm not sure I understand what you mean by a build-system unit test in this context? What behavior do you want to test?

Comment (davezarzycki) [reply test/lit.cfg]: @harlanhaskins – Based on @Rostepher's reply, what do you think about merging this now and worrying about how to unit test the build script later?

Comment (harlanhaskins) [reply test/lit.cfg]: I'll go ahead and merge it and follow up offline with @Rostepher about how to build this unit test.

**Comment by harlanhaskins** [comment]
@swift-ci please clean test

**Comment by harlanhaskins** [comment]
Will merge once this final CI run passes.

**Pull request merged**